src/presenter/films-presenter.js

```diff
@@ -95,14 +128,42 @@ export default class FilmsPresenter {
     this.#cardFilmPresenters.set(filmModelCard.id, cardFilmPresenter);
   }
 
+  #renderFilms(from, to, container) {
+    this.#filmsModelContainer
+      .slice(from, to)
+      .forEach((filmModelCard) => this.#renderFilm(filmModelCard, container));
+  }
+
+  #renderShowMoreButton() {
+    this.#showMoreButtonComponent = new ShowMoreButtonView({
+      onClick: this.#handleShowMoreButtonClick
+    });
+    render(this.#showMoreButtonComponent, this.#filmList.element);
+  }
+
+  #clearFilmList() {
+    this.#cardFilmPresenters.forEach((presenter) => presenter.destroy());
+    this.#cardFilmPresenters.clear();
+    this.#renderedFilmModelCard = FILM_CARD_PER_STEP;
+    remove(this.#showMoreButtonComponent);
+  }
+
+  #renderFilmList() {
+    render(this.#filmListComponent, this.#filmList.element);
+    this.#renderFilms(0, Math.min(this.#filmsModelContainer.length, FILM_CARD_PER_STEP), this.#filmListComponent.element);
+
+    if (this.#filmsModelContainer.length > FILM_CARD_PER_STEP) {
+      this.#renderShowMoreButton();
+    }
+  }
 
   #renderTopAndMost(){
 
     render(this.#topRatedList, this.#filmComponent.element);
     render(this.#topRatedContainer, this.#topRatedList.element);
 
-    this.#mostRatedFilms = getTopRatedFilms(this.#filmsModelContainer);
-    this.#mostCommentedFilms = getMostCommentedFilms(this.#filmsModelContainer);
+    this.#mostRatedFilms = this.#filmsModelContainer.sort(sortByRating).slice(0, 2);
+    this.#mostCommentedFilms = this.#mostCommentedFilms = this.#filmsModelContainer.sort(sortByCommented).slice(0, 2);
```

Comment:
```suggestion
    this.#mostCommentedFilms = this.#filmsModelContainer.sort(sortByCommented).slice(0, 2);
```
Зачем то два разы выполняется присваивание в `this.#mostCommentedFilms`.

И ты же помнишь что сортировка выполняется на месте?
Если вызвать метод `#renderTopAndMost` перед `#renderFilms` то в результате фильмы будут отсортированы по количеству комментариев.

Ещё у тебя есть константа `TOP_AND_MOST_COUNT`.